src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/DefaultLSPDocumentSynchronizer.cs

```diff
@@ -90,7 +92,7 @@ internal record SynchronizedResult<TVirtualDocumentSnapshot>(bool Synchronized,
             }
 
             // Currently tracked synchronizing context is not sufficient, need to update a new one.
-            onSynchronizedTask = documentContext.GetSynchronizationTaskAsync(requiredHostDocumentVersion, rejectOnNewerParallelRequest, cancellationToken);
+            onSynchronizedTask = documentContext.GetSynchronizationTaskAsync(requiredHostDocumentVersion, rejectOnNewerParallelRequest, cancellationToken, isResolveCodeActionSync);
```

Comment:
Does this need to be passed all the way through? It looks to me like the virtual document snapshot will be returned, even if this returns false. So, it seems to me that it'd be better to just put the workaround in `ProvideResolvedCompletionItemAsync` rather than letting it sprawl into the document synchronization, which is already pretty complex.

Reply:
Good point. I've updated this.